core/src/lib.rs

```diff
@@ -1,16 +1,45 @@
 use serde::{Deserialize, Serialize};
 
+#[derive(Debug, Serialize, Deserialize)]
+pub struct PublicKey {
+    pub key: Vec<u8>,
+    pub key_type: String,
```

Comment:
What is the role of `key_type` here?

Reply:
The `key_type` is a string which determines whether the public key is a RSA public key or ED25519 public key.

- Function where the key is used: https://github.com/risc0-labs/dkim/blob/3213315e41f426c307b8feb8fb329330a47e326c/src/lib.rs#L356
- How the key type is determined: https://github.com/zkemail/r0-zkEmail/blob/feat/regex/host/src/utils.rs#L85-L93